### PR TITLE
Prevent hack & slash when avatar.dead

### PR DESCRIPTION
**Pull request by ipdae**
solved #52 

- prevent hack_and_slash move when avatar is dead
- disable find monster button when avatar is dead
- active sleep tab instead of hack & slash when avatar is dead

image

### CODE REVIEW

File: nekoyume/models.py [hunk 1]
```diff
@@ -1353,6 +1353,10 @@ def foods(self) -> list:
 
         return result
 
+    @property
+    def is_dead(self) -> bool:
+        return bool(self.hp <= 0)
```
Comment (longfin) [inline nekoyume/models.py]: Can we write this without explicit `bool`?

File: tests/game_test.py [hunk 2]
```diff
@@ -74,3 +78,22 @@ def test_get_unconfirmed_move(fx_session, fx_user, fx_novice_status):
...
+        fx_private_key: PrivateKey, fx_novice_status: typing.Dict[str, str],
+):
+    move = fx_user.create_novice(fx_novice_status)
+    fx_user.create_block([move])
+
+    assert fx_user.avatar().is_dead is False
+    while fx_user.avatar().hp > 0:
+        move = fx_user.hack_and_slash()
+        fx_user.create_block([move])
+    assert fx_user.avatar().is_dead is True
```
Comment (longfin) [inline tests/game_test.py]: I think `is True` is unnecessary.

File: tests/game_test.py [hunk 3]
```diff
@@ -74,3 +78,22 @@ def test_get_unconfirmed_move(fx_session, fx_user, fx_novice_status):
     move.id = ('00000000000000000000000000000000'
                '00000000000000000000000000000000')
     assert not get_unconfirmed_move(fx_user.address)
+
+
+def test_prevent_hack_and_slash_when_dead(
+        fx_test_client: Client, fx_session: Session, fx_user: User,
```
Comment (longfin) [inline tests/game_test.py]: Actually `Client` is not `werkzeug.test.Client`, but `flask.testing.FlaskClient`.

Comment (ipdae) [reply tests/game_test.py]: 👌

**Pull request merged**